### PR TITLE
8237505: RadioMenuItem in ToggleGroup: deselected on accelerator

### PR DESCRIPTION
**Pull request by karthikpandelu**
No check was present in `RadioMenuItem` accelerator to see if it is in a `ToggleGroup` or not.

Made changes to check if `RadioMenuItem` is part of `ToggleGroup` or not. If it is part of a `ToggleGroup`, then it can not be cleared using accelerator.

Added test to validate the change.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8237505](https://bugs.openjdk.org/browse/JDK-8237505): RadioMenuItem in ToggleGroup: deselected on accelerator

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - Committer)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/1002/head:pull/1002` \
`$ git checkout pull/1002`

Update a local copy of the PR: \
`$ git checkout pull/1002` \
`$ git pull https://git.openjdk.org/jfx pull/1002/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1002`

View PR using the GUI difftool: \
`$ git pr show -t 1002`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1002.diff">https://git.openjdk.org/jfx/pull/1002.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java [hunk 1]
```diff
@@ -175,7 +175,12 @@ private static void doAcceleratorInstall(final List<? extends MenuItem> items, f
                         }
                         if (!menuitem.isDisable()) {
                             if (menuitem instanceof RadioMenuItem) {
-                                ((RadioMenuItem)menuitem).setSelected(!((RadioMenuItem)menuitem).isSelected());
+                                if(((RadioMenuItem)menuitem).getToggleGroup() == null){
```
Comment (andy-goryachev-oracle) [inline modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java:178]: minor: this group insists on adding spaces after "if" and before "{"

**Comment by andy-goryachev-oracle** [comment]
This looks good, given the decision to not to allow deselection by an accelerator.

But I wonder if the right approach is to have a property in the ToggleGroup which determines whether to allow all items to be deselected.  If set, the radio menu items should be allowed to be deselected by both keyboard and mouse; if not set, the first item added to a group should be automatically selected, and radio menu items would not get deselected by the mouse or keyboard.

I think it might qualify as a separate enhancement (and a welcome one, since it saves some boilerplate code).

**Review by andy-goryachev-oracle — APPROVED** [review]
Tested with the MonkeyTester app.

I'd like to have a discussion on a new ToggleGroup's property (i'll send an email to the mailing list).

File: modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java (hunk 1 above)
Comment (mstr2) [reply modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java:178]: You can use a pattern variable here to get rid of all the casts.

Comment (karthikpandelu) [reply modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java]: Updated the code to address above comments

**Comment by karthikpandelu** [comment]
> Tested with the MonkeyTester app.
> 
> I'd like to have a discussion on a new ToggleGroup's property (i'll send an email to the mailing list).

I saw the mail chain sent out for this discussion. I agree that  if any changes required as a result of this discussion can be taken as separate enhancement as you mentioned in the previous comment.

**Review by andy-goryachev-oracle — APPROVED** [review]
lgtm

**Review by aghaisas — APPROVED** [review]
The fix looks good to me!

**Comment by karthikpandelu** [comment]
@aghaisas please sponsor the PR.

**Pull request closed without merging**